crates/re_viewport/src/viewport_blueprint.rs

```diff
@@ -265,79 +297,44 @@ impl<'a> ViewportBlueprint<'a> {
+        if self.auto_layout != value {
+            let component = AutoLayout(value);
+            ctx.save_blueprint_component(&VIEWPORT_PATH.into(), component);
         }
+    }
 
-        if after.auto_layout != before.auto_layout {
-            let component = AutoLayout(after.auto_layout);
-            add_delta_from_single_component(&mut deltas, &entity_path, &timepoint, component);
-        }
-
-        if after.auto_space_views != before.auto_space_views {
-            let component = AutoSpaceViews(after.auto_space_views);
-            add_delta_from_single_component(&mut deltas, &entity_path, &timepoint, component);
-        }
-
-        if after.maximized != before.maximized {
-            let component = SpaceViewMaximized(after.maximized.map(|id| id.into()));
-            add_delta_from_single_component(&mut deltas, &entity_path, &timepoint, component);
-        }
-
-        if after.tree != before.tree {
-            re_log::trace!("Syncing tree");
-
-            let component = ViewportLayout(after.tree.clone());
-
-            add_delta_from_single_component(&mut deltas, &entity_path, &timepoint, component);
+    #[inline]
+    pub fn set_auto_space_views(&self, value: bool, ctx: &ViewerContext<'_>) {
+        if self.auto_layout != value {
+            let component = AutoSpaceViews(value);
+            ctx.save_blueprint_component(&VIEWPORT_PATH.into(), component);
         }
+    }
 
-        // Add any new or modified space views
-        for id in after.space_view_ids() {
-            if let Some(space_view) = after.space_view(id) {
-                sync_space_view(&mut deltas, space_view, before.space_view(id));
-            }
+    #[inline]
+    pub fn set_maximized(&self, space_view_id: Option<SpaceViewId>, ctx: &ViewerContext<'_>) {
+        if self.maximized != space_view_id {
+            let component = SpaceViewMaximized(space_view_id.map(|id| id.into()));
+            ctx.save_blueprint_component(&VIEWPORT_PATH.into(), component);
         }
+    }
 
-        // Remove any deleted space views
-        for space_view_id in before.space_view_ids() {
-            if after.space_view(space_view_id).is_none() {
-                clear_space_view(&mut deltas, space_view_id);
-            }
+    #[inline]
+    pub fn set_tree(&self, tree: &egui_tiles::Tree<SpaceViewId>, ctx: &ViewerContext<'_>) {
+        if &self.tree != tree {
+            re_log::trace!("Updating the layout tree");
+            let component = ViewportLayout(tree.clone());
+            ctx.save_blueprint_component(&VIEWPORT_PATH.into(), component);
         }
-
-        command_sender.send_system(SystemCommand::UpdateBlueprint(
-            after.blueprint_db.store_id().clone(),
-            deltas,
-        ));
     }
 }
 
 // ----------------------------------------------------------------------------
 
 // TODO(jleibs): Move this helper to a better location
-fn add_delta_from_single_component<'a, C>(
+pub fn add_delta_from_single_component<'a, C>(
```

Comment:
It's now only used by `SpaceViewBlueprint::save_full`, so the better place could be to inline it there